### PR TITLE
[jaeger] Added LoadBalancerIP in the Helm Chart

### PR DESCRIPTION
**Pull request by AakashMasand**
I noticed that there is no way to add a LoadBalancerIP to a Helm chart, if someone would want to deploy the Jaeger-collector service with the type Load Balancer. Hence, I have added an optional parameter for LoadBalancerIP that can be specified with the LoadBalancer Jaeger-collector service.

### CODE REVIEW

File: charts/jaeger/templates/collector-svc.yaml [hunk 1]
```diff
@@ -46,5 +46,8 @@ spec:
     {{- include "jaeger.selectorLabels" . | nindent 4 }}
     app.kubernetes.io/component: collector
   type: {{ .Values.collector.service.type }}
+{{- if and (eq .Values.collector.service.type "LoadBalancer") .Values.collector.service.loadBalancerIP }}
+    LoadBalancerIP: {{ .Values.collector.service.type.loadBalancerIP }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/collector-svc.yaml]:
```suggestion
    loadBalancerIP: {{ .Values.collector.service.loadBalancerIP }}
```

**Comment by naseemkullah** [comment]
Thanks @AakashMasand, please bump the chart version, GPG sign your commits and add this to the README configurable values table.

File: charts/jaeger/templates/collector-svc.yaml (hunk 1 above)
Comment (naseemkullah) [reply charts/jaeger/templates/collector-svc.yaml]: loadBalancer, should start with lowercase here as per suggestion ⬆️

**Comment by naseemkullah** [comment]
You need to GPG sign all your commits.

File: charts/jaeger/README.md [hunk 2]
```diff
@@ -262,6 +262,7 @@ The following table lists the configurable parameters of the Jaeger chart and th
 | `collector.service.annotations` | Annotations for Collector SVC | `nil` |
 | `collector.service.grpc.port` | Jaeger Agent port for model.proto | `14250` |
 | `collector.service.http.port` | Client port for HTTP thrift | `14268` |
+| `collector.service.loadBalancerIP` | IP Address to be assigned to the Load Balancer | `nil` |
```
Comment (naseemkullah) [inline charts/jaeger/README.md]: nit: "load balancer" does not require uppercase as they are common nouns in this case.

Maybe just reuse the description you put in the values file?

File: charts/jaeger/templates/collector-svc.yaml (hunk 1 above)
Comment (AakashMasand) [reply charts/jaeger/templates/collector-svc.yaml]: Sure, Ill make the change as well.

File: charts/jaeger/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: 1.17.0
+appVersion: 1.18.0
```
Comment (naseemkullah) [inline charts/jaeger/Chart.yaml]: Please revert this, `appVersion` refers to the version of Jaeger, what you want to bump up is `version`.

Comment (AakashMasand) [reply charts/jaeger/Chart.yaml]: Thanks, @naseemkullah . I will fix this

File: charts/jaeger/README.md (hunk 2 above)
Comment (AakashMasand) [reply charts/jaeger/README.md]: Done :)

File: charts/jaeger/Chart.yaml (hunk 3 above)
Comment (AakashMasand) [reply charts/jaeger/Chart.yaml]: Bumped up chart version to 0.23.0

**Comment by naseemkullah** [comment]
LGTM 👍 ... just need to fix up the conflict and will merge

**Comment by AakashMasand** [comment]
> LGTM 👍 ... just need to fix up the conflict and will merge

Unfortunately, I do not seem to have access to fix it :( . the Mark as Resolved seems to be greyed out.

**Comment by naseemkullah** [comment]
Did you fix this conflict before trying to press `Mark As Resolved`? 
image

**Comment by AakashMasand** [comment]
> Did you fix this conflict before trying to press `Mark As Resolved`?
> image

Thanks, I did not know what was needed exactly. You're a legend, mate :)

**Comment by naseemkullah** [comment]
;)

**Comment by naseemkullah** [comment]
> You need to GPG sign all your commits.

**Comment by AakashMasand** [comment]
> > You need to GPG sign all your commits.

I tried doing this -  git rebase --exec 'git commit --amend --no-edit -n -S' -i development, but did not work

File: charts/jaeger/templates/collector-svc.yaml [hunk 4]
```diff
@@ -46,5 +46,8 @@ spec:
     {{- include "jaeger.selectorLabels" . | nindent 4 }}
     app.kubernetes.io/component: collector
   type: {{ .Values.collector.service.type }}
+{{- if and (eq .Values.collector.service.type "LoadBalancer") .Values.collector.service.loadBalancerIP }}
+    loadBalancerIP: {{ .Values.collector.service.loadBalancerIP }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/collector-svc.yaml]:
```suggestion
  loadBalancerIP: {{ .Values.collector.service.loadBalancerIP }}
```

I think indentation is wrong here. Could you please test locally)?

Comment (AakashMasand) [reply charts/jaeger/templates/collector-svc.yaml]: OK, I will test this out.

**Comment by naseemkullah** [comment]
Hi @AakashMasand did you plan on fixing remaining issues?

**Comment by naseemkullah** [comment]
Closing due to inactivity  @AakashMasand, feel free to comment if still working on this and I will re open. Cheers.

**Pull request closed without merging**